### DISCUSSION TITLE
FIDO2 Server Conformance Testing with the Conformance Tool v1.7.11 or recent

### DISCUSSION THREAD

**Discussion by ThaminduDilshan** [webauthn4j/webauthn4j#819] [General]
Hi @ynojima,

Following up with the discussion on my [previous thread](https://github.com/webauthn4j/webauthn4j/discussions/651), have you run FIDO conformance tests on the tool `v1.7.11` or latest? 

It seems following two issues are now fixed and wondering webauthn4j library could pass all the test cases.

- https://github.com/fido-alliance/conformance-test-tools-resources/issues/618
- https://github.com/fido-alliance/conformance-test-tools-resources/issues/698

Thank you in advance

**Comment by ynojima** [comment]
Now webauthn4j passes these two test cases, but still fails with "P-3 Send a valid ServerAuthenticatorAttestationResponse with FULL "packed" attestation that contains batch certificate, that is simply self referenced in the metadata, and check that server succeeds".
Will fix soon.

![image](https://github.com/webauthn4j/webauthn4j/assets/2139672/ac10c9cd-5bb5-4195-ad9b-294f0d26b443)

Other than that, Metadata Service Tests are also broken, but it seems it is caused by MDS3 test server end.

**Comment by ThaminduDilshan** [comment]
Did you check with the latest suite version? As per the [comment](https://github.com/fido-alliance/conformance-test-tools-resources/issues/698#issuecomment-1466063263), MDS tests are fixed in `v1.7.10` and latest version available is `v1.7.14`

**Reply by ynojima** [reply]
Oh, I tested with `v1.7.5`. Since it is in the experimental directory, I thought it is the latest.
Now I try again with `v1.7.14`, and Metadata Service Tests end with success.

![image](https://github.com/webauthn4j/webauthn4j/assets/2139672/d7c82b33-ff22-45e1-8bef-243e0ce3e18a)

But, another test "P-2 Send a valid ServerAuthenticatorAttestationResponse with "tpm" attestation for SHA-1, and check that server succeeds" start failing. It might be conformance tools' issue, but not sure at this moment.

**Reply by ynojima** [reply]
I found the same issue: https://github.com/fido-alliance/conformance-test-tools-resources/issues/709

**Reply by ThaminduDilshan** [reply]
Thank you very much for the update. Hope it will get resolved soon

**Reply by ynojima** [reply]
Fixed the test case "P-3 Send a valid ServerAuthenticatorAttestationResponse with FULL "packed" attestation that contains batch certificate, that is simply self referenced in the metadata, and check that server succeeds" in https://github.com/webauthn4j/webauthn4j/pull/820, which will be shipped in the next release.

Now the "P-2 Send a valid ServerAuthenticatorAttestationResponse with "tpm" attestation for SHA-1, and check that server succeeds" is the last remaining test case.
![image](https://github.com/webauthn4j/webauthn4j/assets/2139672/1463887b-f4f9-44ff-bfac-dda09f448a9f)

**Reply by ThaminduDilshan** [reply]
Hi,

Test tool version `v1.7.15` is released

**Comment by ynojima** [comment]
With v1.7.15, webauthn4j passed all the test cases.
![image](https://github.com/webauthn4j/webauthn4j/assets/2139672/929f4292-8f39-4aef-bb0d-c907b7d65f23)
